DOC-4744 updated SQL Server preparation guide (#1073)
redis/docs: content/integrate/redis-data-integration/reference/config-yaml-reference.md

### Properties

| Name | Type | Description |
| -- | -- | -- |
| [`sources`](#sources) | `object` | Source collectors |
| [`processors`](#processors)| `object`, `null` | RDI Processors |
| [`targets`](#targets) | `object` | Target connections |

## sources: Source collectors {#sources}

Each source database type has its own connector, but the basic configuration properties are
the same for all databases.

See the Debezium documentation for more information about the specific connectors:

- [MySQL/MariaDB](https://debezium.io/documentation/reference/stable/connectors/mysql.html)
- [Oracle](https://debezium.io/documentation/reference/stable/connectors/oracle.html)
- [PostgreSQL](https://debezium.io/documentation/reference/stable/connectors/postgresql.html)
- [SQL Server](https://debezium.io/documentation/reference/stable/connectors/sqlserver.html)

### Essential properties

#### `connection`

|Name|Type|Source Databases|Description|
|--|--|--|--|
| `host` | `string` | MariaDB, MySQL, Oracle, PostgreSQL, SQLServer|The address of the database instance.|
| `port` | `integer` | MariaDB, MySQL, Oracle, PostgreSQL, SQLServer | The port of the database instance.|

> | `database` | `string` | MariaDB, MySQL, Oracle, PostgreSQL, SQLServer|The name of the database from which to stream the changes. For `SQL Server` you can define the database as comma-separated list of the SQL Server database names from which to stream the changes.|

Reviewer: ```suggestion
| `database` | `string` | MariaDB, MySQL, Oracle, PostgreSQL, SQLServer|The name of the database from which to stream the changes. For `SQL Server` you can define the database as a comma-separated list of the SQL Server database names from which to stream the changes.|
```

| `database.pdb.name` | `string` | Oracle|The name of the [Oracle Pluggable Database](https://docs.oracle.com/en/database/oracle/oracle-database/19/riwin/about-pluggable-databases-in-oracle-rac.html) that the connector captures changes from. For non-CDB installation, do not specify this property.<br/> Default: "ORCLPDB1"|
| `database.encrypt` | `string` | MySQL|If SSL is enabled for a SQL Server database, enable SSL by setting the value of this property to true.<br/> Default: false |
| `database.server.id` | `integer` | MySQL|A numeric ID of this database client, which must be unique across all currently-running database processes in the MySQL cluster.<br/> Default: 1|

> | `database.url` | `string` | Oracle|Specifies the raw database JDBC URL. Use this property to provide flexibility in defining that database connection. Valid values include raw TNS names and RAC connection strings.|

Reviewer: ```suggestion
| `database.url` | `string` | Oracle|Specifies the raw database JDBC URL. Use this property to define a custom database connection. Valid values include raw TNS names and RAC connection strings.|
```

| `topic.prefix` | `string` | MariaDB, MySQL, Oracle, PostgreSQL, SQLServer | A prefix for all topic names that receive events emitted by this connector.<br/> Default: "rdi" |

### Advanced properties

#### `sink`

|Name|Type|Description|
|--|--|--|

> | `redis.null.key` | `string` | Redis does not support the notion of data without key, so this string will be used as key for records without primary key.<br/> Default: "default" |
> | `redis.null.value` | `string` | Redis does not support the notion of null payloads, as is the case with tombstone events. This string will be used as value for records without a payload.<br/> Default: "default" |

Reviewer: ```suggestion
| `redis.null.key` | `string` | Redis does not support the notion of data without key, so this string will be used as the key for records without primary key.<br/> Default: "default" |
| `redis.null.value` | `string` | Redis does not support the notion of null payloads, as is the case with tombstone events. This string will be used as the value for records without a payload.<br/> Default: "default" |
```

 WHERE inv.total > 8000
```

### Form custom message key(s) for change event records

- By default, Debezium uses the primary key column(s) of a table as the message key for records that it emits.
 In place of the default, or to specify a key for tables that lack a primary key, you can configure custom message keys based on one or more columns.

- To establish a custom message key for a table, list the table followed by the column to use as the message key. Each list entry takes the following format:

 ```yaml
 # To include entries for multiple tables, simply add each table with its corresponding columns and keys under the 'tables' field.
 tables:
 <DATABASE_NAME>.<TABLE_NAME>:
 columns:
 - <COLUMN(S)_LIST> # List of columns to include
 keys:
 - <COLUMN(S)_LIST> # Column(s) to be used as the primary key
 ```

 Notes:

 - When specifying columns in the `keys` field, ensure that these same columns are also listed under the `columns` field in your configuration.
 - There is no limit to the number of columns that can be used to create custom message keys. However, it’s best to use the minimum required number of columns to specify a unique key.

### Fully-qualified table name

> In this document we refer to the fully-qualified table name as `<databaseName>.<tableName>`. This format is for MySQL database. For Oracle, SQLServer and Postgresql databases use `<schemaName>`.`<tableName>` instead.

Reviewer: ```suggestion
In this document the fully-qualified table name is referred to as `<databaseName>.<tableName>`. This format is for MySQL database. For Oracle, SQLServer and Postgresql databases use `<schemaName>`.`<tableName>` instead.
```
